search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQARequestProcessor.java

```diff
@@ -19,29 +19,37 @@
 
 import org.opensearch.action.search.SearchRequest;
 import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.ml.common.exception.MLException;
 import org.opensearch.search.pipeline.AbstractProcessor;
 import org.opensearch.search.pipeline.Processor;
 import org.opensearch.search.pipeline.SearchRequestProcessor;
 
 import java.util.Map;
+import java.util.function.BooleanSupplier;
 
 /**
  * Defines the request processor for generative QA search pipelines.
  */
 public class GenerativeQARequestProcessor extends AbstractProcessor implements SearchRequestProcessor {
 
     private String modelId;
+    private final BooleanSupplier featureFlagSupplier;
 
-    protected GenerativeQARequestProcessor(String tag, String description, boolean ignoreFailure, String modelId) {
+    protected GenerativeQARequestProcessor(String tag, String description, boolean ignoreFailure, String modelId, BooleanSupplier supplier) {
         super(tag, description, ignoreFailure);
         this.modelId = modelId;
+        this.featureFlagSupplier = supplier;
     }
 
     @Override
     public SearchRequest processRequest(SearchRequest request) throws Exception {
 
         // TODO Use chat history to rephrase the question with full conversation context.
 
+        if (!featureFlagSupplier.getAsBoolean()) {
+            throw new RuntimeException(GenerativeQAProcessorConstants.FEATURE_NOT_ENABLED_ERROR_MSG);
```

Comment:
Make this also as MLException as you have done in the 87-88 of this class.

Reply:
Ah I missed this one!  Will do.